config.rdf

```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0"?>
+<?xml-stylesheet type="text/xsl" href="owl2html.xsl"?>
+<rdf:RDF 
+     xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#"
+     xmlns:owl="http://www.w3.org/2002/07/owl#"
+     xmlns:xsd="http://www.w3.org/2001/XMLSchema#"
+     xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
+
+    <owl:Ontology rdf:about="http://fedora.info/definitions/v4/config#">
+        <rdfs:label xml:lang="en">Fedora Commons Configuration Ontology</rdfs:label>
+        <rdfs:comment xml:lang="en">Ontology for user-defined configuration properties and classes on Fedora-curated resources.</rdfs:comment>
+    </owl:Ontology>
+
+    <owl:DatatypeProperty rdf:about="http://fedora.info/definitions/v4/config#hasIndexingTransformation">
+        <rdfs:label xml:lang="en">has indexing transformation</rdfs:label>
+        <rdfs:domain rdf:resource="http://fcrepo.info/definitions/v4/repository#Container"/>
```

Comment:
There seems to be some confusion around the idea that a single-namespace
approach will create less work for clients. It won't. In any event, no one
will have to check or use more than one namespace.

More importantly, the single-namespace approach is wrong. It misunderstands
the upcoming spec'ing and partitioning of the API, and the concomitant
approach to the use of ontology. The predicates involved in this example
(the indexing stuff) have literally _nothing_ to do with the operation of
any repository. _Nothing_. They concern _only_ the operation of the
indexing apparatus. So they are naturally associated _to that machinery_.
But there's more to be said.

More broadly, the only ontologies that should be showing up in this
repository are those that would be reused by a completely fresh
implementation of the Fedora repository architecture. (E.g. at UCSD or
Oxford.) It's great that we want to use strong declarative languages like
RDFS and OWL to explain configuration for our implementation. I'm all in
favor of that. But those ontologies don't belong here. They belong in our
implementation, under a namespace specific to our implementation.
(Obviously, they should be _published_ elsewhere, but I'm talking about
where they are maintained.) The indexing predicates should be maintained,
similarly, _with the indexing machinery code_. (Unless we expect to
standardize an indexing SPI to complement the repository APIs.)

Putting a "config" namespace here confuses the Fedora repository
architecture with our implementation thereof, and putting the indexing
predicates into it confuses the repository with the indexer. We can't
expect our community to develop any understanding if we keep confusing them
like that.



On Fri, Jan 16, 2015 at 7:54 PM, Aaron Coburn <notifications@github.com>
wrote:

> In config.rdf
> <https://github.com/fcrepo4/ontology/pull/13#discussion-diff-23119962>:
>
> > +<?xml version="1.0"?>
> > +<?xml-stylesheet type="text/xsl" href="owl2html.xsl"?>
> > +<rdf:RDF
> > +     xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#"
> > +     xmlns:owl="http://www.w3.org/2002/07/owl#"
> > +     xmlns:xsd="http://www.w3.org/2001/XMLSchema#"
> > +     xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
> > +
> > +    <owl:Ontology rdf:about="http://fedora.info/definitions/v4/config#">
> > +        <rdfs:label xml:lang="en">Fedora Commons Configuration Ontology</rdfs:label>
> > +        <rdfs:comment xml:lang="en">Ontology for user-defined configuration properties and classes on Fedora-curated resources.</rdfs:comment>
> > +    </owl:Ontology>
> > +
> > +    <owl:DatatypeProperty rdf:about="http://fedora.info/definitions/v4/config#hasIndexingTransformation">
> > +        <rdfs:label xml:lang="en">has indexing transformation</rdfs:label>
> > +        <rdfs:domain rdf:resource="http://fcrepo.info/definitions/v4/repository#Container"/>
>
> Well, please don't decide on the single namespace approach just because
> there is an existing pull request. The associated code changes in
> fcrepo-camel and fcrepo-message-consumer are truly minimal.
>
> I would only add that both Stefano and I have use cases for the
> *:Indexable class and *:hasIndexingTransform property, and the PR was
> primarily an attempt to begin a discussion so that we can move things along.
>
> @ajs6f <https://github.com/ajs6f> I am not particularly familiar with the
> use of the owl:imports <http://www.w3.org/TR/owl-ref/#imports-def>
> statement. If A imports B, does that mean that all of the classes and
> properties of B are now available under the namespace of A?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fcrepo4/ontology/pull/13/files#r23119962>.
>